### DISCUSSION TITLE
Enable check style for the project

### DISCUSSION THREAD

**Discussion by RobertIndie** [apache/streampipes#815] [Ideas]
Hi all, I would like to use this discussion to discuss enabling check style for all modules in the project.

I created the first PR for demonstration: https://github.com/apache/streampipes/pull/814 

Please feel free to share your thought. Thanks.

**Comment by tenthe** [comment]
Hi, thanks for starting the discussion and providing the PR.
I checked it out and everything worked as expected in my setup. 

+1 to enable it module by module. 
Do you know if it is possible to add the modules to the checkstyle step by step in the maven validation, or do we need to enable it for the whole project at once?
Maybe we can create an issue with subtasks for all the modules so we can work together.

As for the indentation and code header changes. I don't have a strict opinion here as long as it's consistent. So I think that's fine.

One small note: I just saw that the artifactId in the pom.xml is **integration** and the module name is **streampipes-integration-tests**. For the other modules we always have the same name for both, maybe we can change that.

**Comment by RobertIndie** [comment]
> Do you know if it is possible to add the modules to the checkstyle step by step in the maven validation, or do we need to enable it for the whole project at once?

We can do it step by step. Actually, whenever we enable checkstyle in one module by doing like this [PR](https://github.com/apache/streampipes/pull/814), then maven will automatically check the codestyle in CI or by running `mvn install`. 

> Maybe we can create an issue with subtasks for all the modules so we can work together.

Great. I created an issue to track them: https://github.com/apache/streampipes/issues/820

> One small note: I just saw that the artifactId in the pom.xml is integration and the module name is streampipes-integration-tests. For the other modules we always have the same name for both, maybe we can change that.

Thanks for your note. Make sense to me. I will create a PR to change that.

Since we have the issue to track, let's keep all subsequent discussions in https://github.com/apache/streampipes/issues/820
